src/ucs/sys/sock.c

```diff
@@ -237,3 +237,60 @@ const char* ucs_sockaddr_str(const struct sockaddr *sock_addr,
 
     return str;
 }
+
+static ucs_status_t ucs_sockaddr_check_family(const struct sockaddr *sa)
+{
+    if ((sa->sa_family != AF_INET) && (sa->sa_family != AF_INET6)) {
+        ucs_error("unknown address family: %d", sa->sa_family);
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    return UCS_OK;
+}
+
+int ucs_sockaddr_is_equal(const struct sockaddr *sa1,
+                          const struct sockaddr *sa2,
+                          ucs_status_t *status_p)
+{
+    ucs_status_t status = UCS_OK;
+    int result          = 0;
+
+    status = ucs_sockaddr_check_family(sa1);
+    if (status != UCS_OK) {
+        goto out;
+    }
+
+    status = ucs_sockaddr_check_family(sa2);
+    if (status != UCS_OK) {
+        goto out;
+    }
+
+    if (sa1->sa_family != sa2->sa_family) {
```

Comment:
we can check this 1st, so no need to ucs_sockaddr_check_family() for both sa1 and sa2

Reply:
yes, we can
but it returns 0 (not equal) and sets `status=UCS_OK` if `sa1::sa_family=AF_UNIX` and `sa2::sa_family=AF_INET` socket address provided (there are tests in gtests)

done